### ISSUE TITLE
Gradle Plugin: Always register tasks

### ISSUE THREAD

**Issue by hfhbd** [google/ksp#1242]
We have a custom gradle plugin that generates code and ksp should include these sources too, so we want to depend on the ksp tasks like `kspKotlin` when using it with Kotlin/JVM.
```kotlin
project.pluginManager.withPlugin("com.google.devtools.ksp") {
  project.tasks.named("kspKotlin") {
...
  }
}
```
Unfortunately, the `kspKotlin` task is only registered too late, I guess with `afterEvaluate`, so using the `withPlugin` block does not work: `No task with name kspKotlin found.`

Reproducer: 
[untitled.zip](https://github.com/google/ksp/files/10205923/untitled.zip)

**Comment by ting-yuan** [comment]
May I know why you need `withPlugin("com.google.devtools.ksp")`? Wouldn't moving `project.tasks.named("kspKotlin")` out of it work for you?

**Comment by hfhbd** [comment]
Nope. The kspKotlin task is defined by the ksp tool. The `withPlugin` block is called after ksp is applied, so the task `kspKotlin` should be available. But it is registered using `afterEvaluate`, so I have to use `afterEvaluate` too which is a bad practice. So if ksp could register the `kspKotlin` task without `afterEvaluate`, this would be perfect.
